scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/EvaluableLazyParameter.scala

```diff
@@ -14,19 +18,87 @@ class EvaluableLazyParameter[T <: AnyRef](
     override val returnType: TypingResult
 ) extends CustomLazyParameter[T] {
 
-  def this(
-      compiledParameter: CompiledParameter,
+  override val evaluate: Evaluate[T] =
+    LazyParameterEvaluator.evaluate(compiledParameter, expressionEvaluator, nodeId, jobData)
+
+}
+
+class SpelTemplateEvaluableLazyParameter[T <: AnyRef](
+    compiledParameter: BaseCompiledParameter,
+    expressionEvaluator: ExpressionEvaluator,
+    nodeId: NodeId,
+    jobData: JobData
+) extends TemplateLazyParameter[T] {
+
+  override val evaluate: Evaluate[T] =
+    LazyParameterEvaluator.evaluate(compiledParameter, expressionEvaluator, nodeId, jobData)
+
+  override def templateExpression: TemplateExpression = compiledParameter.expression match {
+    case expression: SpelExpression =>
+      expression.templateSubexpressions match {
+        case Some(subexpressions) =>
+          val templateParts = subexpressions.map {
+            case Placeholder(expression) => {
+              new TemplateExpressionPart.Placeholder {
+                override val evaluate: Evaluate[String] = context => {
+                  expressionEvaluator.evaluate[String](expression, "expressionId", nodeId.id, context)(jobData).value
+                }
+              }
+            }
+            case Literal(value) => TemplateExpressionPart.Literal(value)
+          }
+          TemplateExpression(templateParts)
+        case None =>
+          throw new IllegalStateException("Non SpEL-template expression received in SpelTemplateLazyParameter")
```

Comment:
We can get rid of this exceptions by moving this logic to factory method and check `templateSubexpressions` instead of expression langage

Reply:
Not relevant after changes in #7174